doc/sphinx/language/gallina-extensions.rst

```diff
@@ -820,25 +825,25 @@ Sections create local contexts which can be shared across multiple definitions.
    appear inside a section are canceled when the section is closed.
 
 .. cmd:: Let @ident @def_body
+         Let Fixpoint @fix_definition {* with @fix_definition }
+         Let CoFixpoint @cofix_definition {* with @cofix_definition }
+   :name: Let; Let Fixpoint; Let CoFixpoint
 
-   This command binds the value :token:`term` to the name :token:`ident` in the
-   environment of the current section. The name :token:`ident` is accessible
-   only within the current section. When the section is closed, all persistent
-   definitions and theorems within it and depending on :token:`ident`
-   will be prefixed by the let-in definition :n:`let @ident := @term in`.
+   These commands behave like :cmd:`Definition`, :cmd:`Fixpoint` and :cmd`CoFixpoint`, except that
+   the declared constant is local to the current section.
+   When the section is closed, all persistent
+   definitions and theorems within it that depend on :token:`ident`
+   will be prefixed with a :n:`term_let` with the same declaration.
+
+   If :n:`@term` is omitted, :n:`@type` is required and Coq enters proof editing mode.
+   This can be used to define a term incrementally, in particular by relying on the :tacn:`refine` tactic.
+   In this case, the proof should be terminated with :cmd:`Defined` in order to define a constant
+   for which the computational behavior is relevant.  See :ref:`proof-editing-mode`.
 
    .. exn:: @ident already exists.
       :name: @ident already exists. (Let)
       :undocumented:
```

Comment:
IMHO there is no point in documenting this error message here since it is the same as in `Definition`, and since this description already heavily references the latter.